### PR TITLE
Additional results

### PR DESCRIPTION
**Pull request by Hugh-OBrien**
3 additions:
## \- Bulk data collector 

An example of how we could produce complete timeline data for the visualisation. This could also be useful for looking closely at our algorithms to see if they're functioning correctly at a granular level. Currently this example follows a single piece of data marked '1' and logs all related requests on the network in the form:

([(requester_node, current_locations_of_content), Path_Taken], Time_Stamp) 

Can be accessed by adding 'BULK_DATA' to the data collectors in the config file

---
## \- Data collector for topology data

This will add the start state of each experiments topology to the results dataset. Warning: will result in large results files if there is a lot of content in the network since the starting point of each piece of content is included in this result

Can be accessed by adding 'TOPOLOGY' to the data collectors in the config file

---
## \- #9 - Support for JSON outputs from Icarus. There are 2 types:
1. 'JSON' - which provides regular results in JSON rather than Pickle using jsonpickle
2. 'JSON_TOPOLOGY' - will also provide individually numbered topology files for each experiment provided the TOPOLOGY data collector is included in the config file

Both are accessible by changing the 'RESULTS_FORMAT' in the config file

### CODE REVIEW

**Comment by Hugh-OBrien** [comment]
@FilWisher I'm not totally sure we agreed on a pull request procedure so I've just assigned you to this. 

@David-Loughnane we can work on improving the usefulness of this data, but there's an example here in /icarus/examples/JSON_Results_Test/Results_Topology_0.json of a flattened set of just the quickly callable node and edge data.

File: #.gitignore# [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Large.json
```
Comment (FilWisher) [inline #.gitignore#]: Could you remove this file from the PR?

File: icarus/examples/JSON_results_test/config.py~ [hunk 2]
```diff
@@ -0,0 +1,70 @@
+"""This module contains all configuration information used to run simulations
```
Comment (FilWisher) [inline icarus/examples/JSON_results_test/config.py~]: Could you remove this file from the PR?

File: icarus/examples/JSON_results_test/run.sh~ [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
```
Comment (FilWisher) [inline icarus/examples/JSON_results_test/run.sh~]: Could you remove this file from the PR?

File: icarus/examples/QuickSim/config.py~ [hunk 4]
```diff
@@ -0,0 +1,126 @@
+"""This module contains all configuration information used to run simulations
```
Comment (FilWisher) [inline icarus/examples/QuickSim/config.py~]: Could you remove this file from the PR?

File: icarus/examples/QuickSim2/visualiseHelper.py~ [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import sys, os
```
Comment (FilWisher) [inline icarus/examples/QuickSim2/visualiseHelper.py~]: Could you remove this file from the PR?

**Comment by FilWisher** [comment]
This is looking good apart from a few editor temp files added by accident. Happy to merge once they are removed.

**Comment by Hugh-OBrien** [comment]
@FilWisher Should be all good now. Sorry about that, my bad

**Pull request merged**